### PR TITLE
vSphere UPI - Use a folder for virtual machines

### PR DESCRIPTION
**Pull request by jcpowermac**
- Creates a vm folder
- Virtual Machines are deployed into the folder created

### CODE REVIEW

**Comment by jcpowermac** [comment]
cc @staebler

File: upi/vsphere/folder/variables.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+variable "path" {
+  type = "string"
+}
+
+variable "datacenter_id" {
+  type = "string"
+}
+
+variable "vsphere_cluster" {
```
Comment (staebler) [inline upi/vsphere/folder/variables.tf]: This variable is not used.

File: upi/vsphere/machine/variables.tf [hunk 2]
```diff
@@ -20,6 +20,10 @@ variable "resource_pool_id" {
   type = "string"
 }
 
+variable "folder_id" {
```
Comment (staebler) [inline upi/vsphere/machine/variables.tf]: I'm not fond of `folder_id` as the name of this variable. It is not an ID. It is the name of the folder. My preference would be for this to just be `folder`. I would be fine with `folder_name` or `folder_path`, too.

**Comment by staebler** [comment]
Can you change the commit message to start with "upi/vsphere" and remove the [WIP] from the commit message?

**Comment by jcpowermac** [comment]
> Changes look good. I have just a couple nits.
> 
> Did using the path from the vsphere_folder resource set up the dependency so that the folder is not deleted until the vms are deleted? Or do we still have the deficiency where terraform attempts to delete the folder too early?

I tested a number of times to make sure that the folder was deleted after the vms.  I wanted to make sure that worked before we went ahead with this PR.

**Comment by staebler** [comment]
> > Changes look good. I have just a couple nits.
> > Did using the path from the vsphere_folder resource set up the dependency so that the folder is not deleted until the vms are deleted? Or do we still have the deficiency where terraform attempts to delete the folder too early?
> 
> I tested a number of times to make sure that the folder was deleted after the vms. I wanted to make sure that worked before we went ahead with this PR.

Great.I have used it a couple times too, with success.

**Pull request merged**